engines/sci/graphics/screen.cpp

```diff
@@ -174,38 +165,78 @@ GfxScreen::GfxScreen(ResourceManager *resMan, Common::RenderMode renderMode) : _
 	bool requestRGB = enablePaletteMods || (ConfMan.hasKey("rgb_rendering") && ConfMan.getBool("rgb_rendering"));
 
 	_gfxDrv = nullptr;
-	if (getSciVersion() <= SCI_VERSION_0_LATE || getSciVersion() == SCI_VERSION_1_EGA_ONLY) {
-		switch (renderMode) {
-		case Common::kRenderCGA:
-			_gfxDrv = new SCI0_CGADriver(false, requestRGB);
-			break;
-		case Common::kRenderCGA_BW:
-			_gfxDrv = new SCI0_CGABWDriver(0xffffff, requestRGB);
-			break;
-		case Common::kRenderHercA:
-		case Common::kRenderHercG:
-			_gfxDrv = new SCI0_HerculesDriver(renderMode == Common::kRenderHercG ? 0x66ff66 : 0xffbf66, requestRGB, false);
-			break;
-		default:
-			break;
-		}
-	} else {
-		switch (renderMode) {
-		case Common::kRenderEGA:
+	switch (renderMode) {
+	case Common::kRenderCGA:
+		_gfxDrv = new SCI0_CGADriver(false, requestRGB);
+		break;
+	case Common::kRenderCGA_BW:
+		_gfxDrv = new SCI0_CGABWDriver(0xffffff, requestRGB);
+		break;
+	case Common::kRenderHercA:
+	case Common::kRenderHercG:
+		_gfxDrv = new SCI0_HerculesDriver(renderMode == Common::kRenderHercG ? 0x66ff66 : 0xffbf66, requestRGB, false);
+		break;
+	case Common::kRenderEGA:
+		// No support for this mode in the Korean version yet.
+		if (getSciVersion() > SCI_VERSION_1_EGA_ONLY && g_sci->getLanguage() != Common::KO_KOR)
 			_gfxDrv = new SCI1_EGADriver(requestRGB);
-			break;
-		case Common::kRenderVGAGrey:
+		break;
+	case Common::kRenderVGAGrey:
+		// No support for this mode in the Korean version yet.
+		if (g_sci->getLanguage() != Common::KO_KOR)
 			_gfxDrv = new SCI1_VGAGreyScaleDriver(requestRGB);
+		break;
+	case Common::kRenderPC98_8c:
+		if (g_sci->getGameId() == GID_PQ2)
+			// PQ2 is a bit special, probably the oldest of the PC-98 ports. Unlike all the others, it uses text mode print
+			// and it doesn't even have a 16 colors drivers. See comment below...
+			_gfxDrv = new SCI0_PC98Gfx8ColorsDriver(true, SCI0_PC98Gfx8ColorsDriver::kFontStyleFat, 1, requestRGB);
+		else if (getSciVersion() <= SCI_VERSION_01)
+			_gfxDrv = new SCI0_PC98Gfx8ColorsDriver(false, SCI0_PC98Gfx8ColorsDriver::kFontStyleNone, -1, requestRGB);
+		else
+			_gfxDrv = new SCI1_PC98Gfx8ColorsDriver(requestRGB);
+		_hiresGlyphBuffer = new byte[16 * 16]();
+		break;
+	default:
+		break;
+	}
+
+	if (_gfxDrv == nullptr) {
+		switch (g_sci->getPlatform()) {
+		/*case Common::kPlatformMacintosh:
```

Comment:
Did you mean to include this?

Reply:
It is kind of what I said in the description about "some skeleton code for Mac, but it is currently useless/disabled". So the answer to your question would be yes. But I can throw it out for sure. I was just wary of messing up my local branches too much. But removing that would actually relieve me of any pressure to deliver anything in the near future, so probably a good idea. I'll add a cleanup commit...